Fix SUSEConnect conflict with zypper

lemon-suse opened the pull request:
After quit the packagekit, it need wait a while to exit related zypper process, then can start the SUSUConnect cmds.

- Related ticket: https://progress.opensuse.org/issues/77134
- Needles: N/A
- Verification run: http://openqa.nue.suse.com/tests/4981864#step/system_prepare/14

okurz commented on tests/console/system_prepare.pm:
```diff
@@ -34,8 +34,12 @@ sub run {
+        # Make sure packagekit is not running, or it will conflict with SUSEConnect.
+        pkcon_quit unless check_var('DESKTOP', 'textmode');
+        # poo#77134 wait for packagekit related zypper process to exit
+        wait_still_screen 10;
```
this is unreliable and should be avoided. If you need to wait for the related zypper processes then do exactly that, e.g.:

```
assert_script_run 'until ! pgrep zypper; do sleep 1; done';
```

lemon-suse replied:
Yes, this is a better way to wait the block zypper process.

okurz commented on tests/console/system_prepare.pm:
```diff
@@ -34,8 +34,12 @@ sub run {
-    pkcon_quit unless check_var('DESKTOP', 'textmode');
+    if (!check_var('DESKTOP', 'textmode')) {
+        # Make sure packagekit is not running, or it will conflict with SUSEConnect.
+        pkcon_quit unless check_var('DESKTOP', 'textmode');
```
the check is redundant now

lemon-suse replied:
Updated code and verification log: https://openqa.nue.suse.com/tests/4991482#step/system_prepare/14

okurz reviewed (changes requested):
typo SUSConnect in subject

b10n1k commented on tests/console/system_prepare.pm:
```diff
@@ -34,8 +34,12 @@ sub run {
+        # Make sure packagekit is not running, or it will conflict with SUSEConnect.
+        pkcon_quit;
+        # poo#77134 wait for packagekit related zypper process to exit
+        assert_script_run 'until ! pgrep zypper; do sleep 1; done';
```
can i suggest use of script_retry https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1493

lemon-suse replied:
Thanks, I think it is correct function to do such things here. I just think both work for me, I will try it next time. Thanks.

okurz replied:
I recommend against that as the recurring switch between the scope of remote-shell and local-perl is more expensive and error prone. `script_retry` can be helpful if something as simple as such loop does not work.

lemon-suse replied:
Thanks for the analysis for the difference for the loop script and the function script_retry. It's very helpful.

lemon-suse commented:
Hi Oliver, do you still have any comments? Thanks.

okurz commented:
yes, still the same which have not changed as the typo was not fixed. And I think you also introduced a new one since then with SUSUConnect. Please take care about correct spelling – not just to present a proper "quality". Correct spelling is also necessary when one wants to look into the git log at a later time to understand changes. And when using a search term like "SUSEConnect", your git commit would not show up when you write SUSConnect or SUSUConnect.

I suggest before asking others to review you should look at your commits from the perspective of another person and try to find out if everything is clearly readable. I know, I also sometimes fail at this or forget it. It is good to have helpful colleagues as a safety net :)

lemon-suse commented:
> yes, still the same which have not changed as the typo was not fixed. And I think you also introduced a new one since then with SUSUConnect. Please take care about correct spelling – not just to present a proper "quality". Correct spelling is also necessary when one wants to look into the git log at a later time to understand changes. And when using a search term like "SUSEConnect", your git commit would not show up when you write SUSConnect or SUSUConnect.
> 
> I suggest before asking others to review you should look at your commits from the perspective of another person and try to find out if everything is clearly readable. I know, I also sometimes fail at this or forget it. It is good to have helpful colleagues as a safety net :)


Thanks Oliver. Sorry or the typo but the reason is for the poor network recently, maybe it cause removed 'E' unexpected.
Anyway, I will take care that later. Updated the commit. Thanks.

okurz commented:
I hope – especially if you have poor network – that you work on git commits on a local machine. Then I can't imagine how network would be involved.

There is still "SUSUConnect" in the git commit message description.

lemon-suse commented:
> I hope – especially if you have poor network – that you work on git commits on a local machine. Then I can't imagine how network would be involved.
> 
> There is still "SUSUConnect" in the git commit message description.



> I hope – especially if you have poor network – that you work on git commits on a local machine. Then I can't imagine how network would be involved.
> 
> There is still "SUSUConnect" in the git commit message description.

Updated the commit. Thanks.
About network, I need connect to my workstation in office for openQA workspace via VPN which is very slow. Anyway, I'm thinking to build workspace on local machine instead.

okurz commented:
I see. For that I recommend "mosh" instead of standard ssh. It helps with flaky and slow connections with "echo prediction" and automatic reconnects.

lemon-suse commented:
> I see. For that I recommend "mosh" instead of standard ssh. It helps with flaky and slow connections with "echo prediction" and automatic reconnects.

Ok, I will try it later. Thanks.
